java/code/src/com/redhat/rhn/common/db/datasource/DataList.java

```diff
@@ -99,7 +99,7 @@ private void elaborate() {
                  * To avoid this, turn off automatic elaboration.
                  */
                 autoElab = false;
-                mode.elaborate(this, elaboratorParams);
+                mode.elaborate((List<Object>)this, elaboratorParams);
```

Comment:
I wonder if we could go with type parametrization. @mackdk any opinion on it?

Reply:
I don't think casting here to `List<Object>` adds any value. `DataList` (the `this` reference) is already parameterized.  What we should do is parameterize `SelectMode::elaborate`, in the same way `SelectMode::execute` is parameterized.

The main problem the problem is that `CachedStatement` (which is used by `SelectMode`) is very convoluted code when it comes to type handling. There are methods there returning either a list or number depending on the input, or accepting both a map or a single value as a parameter. So, it's impossible to achieve real type safety there without a proper refactor.

The same holds true for the changes in `DataResult`.

Reply:
@RohitBhojak any update on this discussion?